Sources/Core/CharcoalViewController.swift

```diff
@@ -49,28 +41,20 @@ public class CharcoalViewController: UINavigationController {
 
     // MARK: - Private properties
 
+    private var filter: FilterContainer?
+    private var config: FilterConfiguration?
+
     private var selectionHasChanged = false
-    private let selectionStore: FilterSelectionStore
-    private var rootFilterViewController: RootFilterViewController
+    private var selectionStore = FilterSelectionStore()
 
+    private var rootFilterViewController: RootFilterViewController?
     private lazy var loadingViewController = LoadingViewController(backgroundColor: .milk, presentationDelay: 0)
 
     // MARK: - Init
 
-    public init(filter: FilterContainer, config: FilterConfiguration, queryItems: Set<URLQueryItem> = []) {
```

Comment:
Could we keep `config` in `init`? Isn't it created only once?

Reply:
When we change submarket we change both the filter and config. We could potentially make one config for all markets and submarkets.